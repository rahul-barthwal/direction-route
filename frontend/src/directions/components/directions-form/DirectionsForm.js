import React, { Component } from 'react';
import { mapClient } from '../../../common/map-client/map-client';

import './DirectionsForm.scss';

/**
 * @name DirectionsForm
 * @type {Component}
 * @extends Component React component
 * @description This component contins the auto complete form
 */
class DirectionsForm extends Component {
    /**@constructor */
    constructor() {
        super();
        // Starting point input, needs to save ref
        this.fromInput = undefined;
        // Starting point Autocomplete, After rendering the google autocomplete, reference will be saved in this
        this.fromInputAutoComplete = undefined;
        // Drop-off point input, needs to save ref
        this.toInputAutoComplete = undefined;
        // Drop-off point Autocomplete, After rendering the google autocomplete, reference will be saved in this
        this.toInput = undefined;
    }

    /**
     * @name getRoute
     * @description call the parent 'getDirections' method and supply current Starting and drop-off points
     */
    getRoute = () => {
        const from = this.fromInputAutoComplete.getPlace();
        const to = this.toInputAutoComplete.getPlace();
        this.props.getDirections(from, to);
    };

    /**
     * @name renderAutoComplete
     * @description Attach the google places autocomplete to the inputs
     */
    renderAutoComplete = async () => {
        const google = await mapClient;

        this.fromInputAutoComplete = new google.maps.places.Autocomplete(
            this.fromInput
        );

        this.toInputAutoComplete = new google.maps.places.Autocomplete(
            this.toInput
        );
    };

    /**
     * Component lifecycle hook
     */
    componentDidMount() {
        this.renderAutoComplete();
    }

    /**
     * Component render method
     */
    render() {
        return (
            <div className="directions-form">
                <div className="form-input">
                    <label>Starting Location</label>
                    <input type="text" ref={el => (this.fromInput = el)} />
                </div>
                <div className="form-input">
                    <label>Drop-off point</label>
                    <input type="text" ref={el => (this.toInput = el)} />
                </div>
                <div className="get-route-btn">
                    <button onClick={this.getRoute}>Get Route</button>
                </div>
            </div>
        );
    }
}

export default DirectionsForm;
