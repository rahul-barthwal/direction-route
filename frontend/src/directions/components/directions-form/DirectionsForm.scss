@import '../../../responsive.scss';

.directions-form {
    padding: 16px;
    display: flex;
    align-items: flex-end;

    @include mobile {
        flex-direction: column;
        align-items: flex-start;
    }
}

.form-input {
    margin-top: 16px;
    margin-right: 16px;

    > label {
        margin-bottom: 12px;
        font-size: 12px;
        display: block;
    }

    > input {
        display: block;
        padding: 8px;
        width: 200px;
    }
}

.get-route-btn > button {
    margin-top: 16px;
    padding: 10px;
    min-width: 100px;
    border: none;
    background: green;
    color: #ffffff;
    cursor: pointer;
}
