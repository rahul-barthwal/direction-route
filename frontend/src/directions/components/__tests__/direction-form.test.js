import React from 'react';
import renderer from 'react-test-renderer';

import DirectionsForm from '../directions-form/DirectionsForm';

describe('Direction Form Test', () => {
    it('render correctly', () => {
        const tree = renderer.create(<DirectionsForm />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
