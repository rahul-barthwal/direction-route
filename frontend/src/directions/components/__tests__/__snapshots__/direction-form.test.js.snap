// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Direction Form Test render correctly 1`] = `
<div
  className="directions-form"
>
  <div
    className="form-input"
  >
    <label>
      Starting Location
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label>
      Drop-off point
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="get-route-btn"
  >
    <button
      onClick={[Function]}
    >
      Get Route
    </button>
  </div>
</div>
`;
